optimum/habana/transformers/generation/utils.py

```diff
@@ -588,8 +589,8 @@ def generate(
             not generation_config.bucket_internal
             and generation_config.bucket_size > 0
             and (
-                self._get_generation_mode(generation_config, assistant_model) == GenerationMode.GREEDY_SEARCH
-                or self._get_generation_mode(generation_config, assistant_model) == GenerationMode.BEAM_SEARCH
+                generation_config.get_generation_mode(assistant_model) == GenerationMode.GREEDY_SEARCH
```

Comment:
This change is compatible with transformer 4.39 but not what we have now. I think we have to wait for transformer upgrade before merging this change.